File: 03-30-2018/csp.rb
# Read input
n, k = gets.split.map(&:to_i)
integers = gets.split.map(&:to_i)

# Count the number of times integers_i + integers_j is divisible by k when i < j
count = 0
(1...integers.length).each do |j|
  (0...j).each do |i|
    count += 1 unless (integers[i] + integers[j]) % k != 0
  end
end

puts count

File: 03-30-2018/csp.py
from functools import reduce
n, k = map(lambda x: int(x), input().split())
numbers = list(map(lambda x: int(x), input().split()))
print(reduce((lambda x, y: x + y), map(lambda val: 1 if (val % k == 0) else 0, [numbers[i]+numbers[j] for j in range(1,n) for i in range(0, j) if i < j])))
